python/ray/util/dask/callbacks.py

```diff
@@ -0,0 +1,211 @@
+import contextlib
+from collections import namedtuple
+
+from dask.callbacks import Callback
+
+# The names of the Ray-specific callbacks. These are the kwarg names that
+# RayDaskCallback will accept on construction, and is considered the
+# source-of-truth for what Ray-specific callbacks exist.
+CBS = (
+    "ray_presubmit",
+    "ray_postsubmit",
+    "ray_pretask",
+    "ray_posttask",
+    "ray_postsubmit_all",
+    "ray_finish",
+)
+# The Ray-specific callback method names for RayDaskCallback.
+CB_FIELDS = tuple("_" + field for field in CBS)
```

Comment:
I would really like to _not_ prefix the Ray-specific callbacks with an underscore and instead make the `RayDaskCallback` constructor kwargs (`RayDaskCallback(ray_presubmit=foo)`) and method implementations (`def ray_presubmit(...):`) have the same identifier, but I think that'd deviate too strongly from Dask's custom callback API.